src/test/java/org/cryptomator/ui/error/ErrorControllerTest.java

```diff
@@ -0,0 +1,252 @@
+				stackTrace,
+				errorCode,
+				previousScene,
+				window,
+				environment,
+				executorService);
+	}
+
+	private ErrorDiscussion createErrorDiscussion(String title, int upvoteCount, ErrorDiscussion.Answer answer){
+		ErrorDiscussion ed = new ErrorDiscussion();
+		ed.title =title;
+		ed.upvoteCount = upvoteCount;
+		ed.answer = answer;
+		return ed;
+	}
+
+	@Test
+	@DisplayName("If the upvoteCounts of both ErrorDiscussions are equal, it returns 0")
+	void testCompareUpvoteCount1() {
+		ErrorDiscussion ed1 = createErrorDiscussion("",10,null);
+		ErrorDiscussion ed2 = createErrorDiscussion("",10,null);
+		int result = errorController.compareUpvoteCount(ed1, ed2);
+		Assertions.assertEquals(0, result);
+	}
+
+
+	@Test
+	@DisplayName("If the upvoteCount of the first ErrorDiscussion is greater than the second, it returns < 0.")
+	void testCompareUpvoteCount2() {
+		ErrorDiscussion ed1 = createErrorDiscussion("",10,null);
+		ErrorDiscussion ed2 = createErrorDiscussion("",5,null);
+		int result = errorController.compareUpvoteCount(ed1, ed2);
+		Assertions.assertTrue(result < 0);
+	}
+
+	@Test
+	@DisplayName("If the upvoteCount of the second ErrorDiscussion is greater than the first, it returns > 0.")
+	void testCompareUpvoteCount3() {
+		ErrorDiscussion ed1 = createErrorDiscussion("",8,null);
+		ErrorDiscussion ed2 = createErrorDiscussion("",15,null);
+		int result = errorController.compareUpvoteCount(ed1, ed2);
+		Assertions.assertTrue(result > 0);
+	}
+
+	@Test
+	@DisplayName("If both ErrorDiscussions has an answer, it returns 0.")
+	void testCompareIsAnswered1() {
+		ErrorDiscussion ed1 = createErrorDiscussion("",0, new ErrorDiscussion.Answer());
+		ErrorDiscussion ed2 = createErrorDiscussion("",0, new ErrorDiscussion.Answer());
+		int result = errorController.compareIsAnswered(ed1,ed2);
+		Assertions.assertEquals( 0, result);
+	}
+
+	@Test
+	@DisplayName("If both ErrorDiscussions doesn't have an answer, it returns 0.")
+	void testCompareIsAnswered2() {
+		ErrorDiscussion ed1 = createErrorDiscussion("",0, null);
+		ErrorDiscussion ed2 = createErrorDiscussion("",0, null);
+		int result = errorController.compareIsAnswered(ed1,ed2);
+		Assertions.assertEquals(result, 0);
+	}
+
+	@Test
+	@DisplayName("If the first ErrorDiscussion has an answer and the second doesn't, it returns < 0.")
+	void testCompareIsAnswered3() {
+		ErrorDiscussion ed1 = createErrorDiscussion("",0, new ErrorDiscussion.Answer());
+		ErrorDiscussion ed2 = createErrorDiscussion("",0,null);
+		int result = errorController.compareIsAnswered(ed1,ed2);
+		Assertions.assertTrue(result < 0);
+	}
+
+	@Test
+	@DisplayName("If the second ErrorDiscussion has an answer and the first doesn't, it returns > 0.")
+	void testCompareIsAnswered4() {
+		ErrorDiscussion ed1 = createErrorDiscussion("",0, null);
+		ErrorDiscussion ed2 = createErrorDiscussion("",0, new ErrorDiscussion.Answer());
+		int result = errorController.compareIsAnswered(ed1,ed2);
+		Assertions.assertTrue(result > 0);
+	}
```

Comment:
Tests with the same structure should be combined into a single parameterized test.

```suggestion
	@ParameterizedTest
	@CsvSource(textBlock = """
		false, false, 0
		true, true, 0 
		true, false, -1 
		false, true, 1 
		""")
	public void testCompareIsAnswered(boolean leftIsAnswered, boolean rightIsAnswered, int expectedResult) {
		var answer = new ErrorDiscussion.Answer();
		var left = createErrorDiscussion("", 0, leftIsAnswered ? answer : null);
		var right = createErrorDiscussion("", 0, rightIsAnswered ? answer : null);
		int result = errorController. compareIsAnswered(left, right);
		Assertions.assertEquals(expectedResult, result);
	}
```

same goes for the other comparisons as well.